### PR TITLE
feat(dashboard): inventory create flow

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- inventory create flow
- SDK methods + HTTP types
- reorg and cleanup

**Fix**
- batch create location levels wasn't working due to improper link removal
- list inventory levels filtering by item id

---

https://github.com/medusajs/medusa/assets/16856471/cfe4ec4e-c7b8-4ae4-8772-d630e6ba967a

---

CLOSES CORE-2233

### CODE REVIEW

File: packages/core/types/src/http/inventory/admin/queries.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { FindParams } from "../../common"
+import { OperatorMap } from "../../../dal"
+
+export interface AdminInventoryItemFilters extends FindParams {
```
Comment (olivermrbl) [inline packages/core/types/src/http/inventory/admin/queries.ts]: nit:
```suggestion
export interface AdminInventoryItemParams extends FindParams {
```

File: packages/core/js-sdk/src/admin/inventory-item.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        method: "POST",
+        headers,
+        body,
+        query,
+      }
+    )
+  }
+
+  async list(
+    query?: FindParams & HttpTypes.AdminInventoryItemFilters,
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/admin/inventory-item.ts]: question: Doesn't the type already extend the `FindParams`?
```suggestion
    query?: FindParams & HttpTypes.AdminInventoryItemFilters,
```

File: packages/core/js-sdk/src/admin/inventory-item.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      `/admin/inventory-items/${id}`,
+      {
+        query,
+        headers,
+      }
+    )
+  }
+
+  async delete(id: string, headers?: ClientHeaders) {
+    return await this.client.fetch<DeleteResponse<"inventory_item">>(
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/admin/inventory-item.ts]: suggestion: Create a delete response type in `inventory-item/admin/responses.ts` for consistency

File: packages/core/js-sdk/src/admin/inventory-item.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      {
+        method: "DELETE",
+        headers,
+      }
+    )
+  }
+
+  async listLevels(
+    id: string,
+    query?: FindParams & HttpTypes.AdminInventoryLevelFilters,
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/admin/inventory-item.ts]: Same as other question

File: packages/core/js-sdk/src/admin/inventory-item.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    return await this.client.fetch<DeleteResponse<"inventory_item">>(
+      `/admin/inventory-items/${id}/location-levels/${locationId}`,
+      {
+        method: "DELETE",
+        headers,
+      }
+    )
+  }
+
+  async batchPostLevels(
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/admin/inventory-item.ts]: suggestion:
```suggestion
  async batchUpdateLevels(
```

File: packages/core/types/src/http/inventory-level/admin/payloads.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+export interface AdminCreateInventoryLevel {
+  location_id: string
+  stocked_quantity?: number
+  incoming_quantity?: number
+}
+
+export interface AdminBatchPostInventoryLevelLocation {
+  create?: AdminCreateInventoryLevel[]
+  update?: AdminUpdateInventoryLevel[]
+  delete?: never // TODO - not implemented
```
Comment (olivermrbl) [inline packages/core/types/src/http/inventory-level/admin/payloads.ts]: question: It looks like you included this functionality in the PR – is something still missing?

**Review by olivermrbl — COMMENTED** [review]
Nice work, left a few comments. Overall, the BE looks good. Will get @kasperkristensen review the FE changes 👍

File: packages/core/js-sdk/src/admin/inventory-item.ts (hunk 2 above)
Comment (fPolic) [reply packages/core/js-sdk/src/admin/inventory-item.ts]: good catch, I will fix it

File: packages/core/types/src/http/inventory-level/admin/payloads.ts (hunk 6 above)
Comment (fPolic) [reply packages/core/types/src/http/inventory-level/admin/payloads.ts]: Good catch, we didn't have delete param explicitly defined but the action is supported. On the other hand, update params are defined in the schema but update action is not supported in the workflow so I updated the type to reflect that.

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 7]
```diff
@@ -57,26 +46,55 @@ export const useInventoryItem = (
   id: string,
   query?: Record<string, any>,
   options?: Omit<
-    UseQueryOptions<InventoryItemRes, Error, InventoryItemRes, QueryKey>,
+    UseQueryOptions<
+      HttpTypes.AdminInventoryItemResponse,
+      Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: suggestion: type as `FetchError` from js-sdk

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 8]
```diff
@@ -57,26 +46,55 @@ export const useInventoryItem = (
...
     ...options,
   })
 
   return { ...data, ...rest }
 }
 
+export const useCreateInventoryItem = (
+  options?: UseMutationOptions<
+    HttpTypes.AdminInventoryItemResponse,
+    Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: same as above

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 9]
```diff
@@ -112,14 +134,15 @@ export const useDeleteInventoryItem = (
 export const useDeleteInventoryItemLevel = (
   inventoryItemId: string,
   locationId: string,
-  options?: UseMutationOptions<InventoryItemDeleteRes, Error, void>
+  options?: UseMutationOptions<
+    HttpTypes.AdminInventoryItemDeleteResponse,
+    Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: same as above

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 10]
```diff
@@ -141,40 +164,35 @@ export const useInventoryItemLevels = (
   query?: Record<string, any>,
   options?: Omit<
     UseQueryOptions<
-      InventoryItemLocationLevelsRes,
+      HttpTypes.AdminInventoryLevelListResponse,
       Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: same as above

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 11]
```diff
@@ -141,40 +164,35 @@ export const useInventoryItemLevels = (
...
 }
 
-export const useUpdateInventoryItemLevel = (
+export const useUpdateInventoryLevel = (
   inventoryItemId: string,
   locationId: string,
   options?: UseMutationOptions<
-    AdminInventoryItemResponse,
+    HttpTypes.AdminInventoryItemResponse,
     Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: same as above

File: packages/admin-next/dashboard/src/hooks/api/inventory.tsx [hunk 12]
```diff
@@ -191,17 +209,17 @@ export const useUpdateInventoryItemLevel = (
   })
 }
 
-export const useBatchInventoryItemLevels = (
+export const useBatchUpdateInventoryLevels = (
   inventoryItemId: string,
   options?: UseMutationOptions<
-    InventoryItemLocationLevelsRes,
+    HttpTypes.AdminInventoryItemResponse,
     Error,
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/api/inventory.tsx]: same as above

File: packages/admin-next/dashboard/src/providers/router-provider/route-map.tsx [hunk 13]
```diff
@@ -468,6 +468,10 @@ export const RouteMap: RouteObject[] = [
                 path: "",
                 lazy: () => import("../../routes/inventory/inventory-list"),
               },
+              {
+                path: "create",
+                lazy: () => import("../../routes/inventory/inventory-create"),
+              },
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/providers/router-provider/route-map.tsx]: should be a child of the list page

File: packages/admin-next/dashboard/src/components/data-grid/data-grid-cells/data-grid-number-cell.tsx [hunk 14]
```diff
@@ -18,7 +23,14 @@ export const DataGridNumberCell = <TData, TValue = any>({
         type="number"
         {...register(field, {
           valueAsNumber: true,
+          onChange: (e) => {
+            if (e.target.value) {
+              return parseInt(e.target.value)
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/components/data-grid/data-grid-cells/data-grid-number-cell.tsx]: We should properly use another function for parsing the value, as this cell might be used for fields that aren't integers

**Review by kasperkristensen — APPROVED** [review]
LGTM, just small things to address before merging

**Pull request merged**